### ISSUE TITLE
Can someone tell me why target signer is paying for Transaction and the source signer is just paying transaction fee?

### ISSUE THREAD

**Issue by arunjot12** [paritytech/parity-bridges-common#2125]
In a cross-chain transaction, the source chain pays the transaction fee, while the target chain pays the execution fee. What thing is  causing the target signer to pay the amount.?

**Comment by serban300** [comment]
There are some details regarding the fees here: https://github.com/paritytech/parity-bridges-common/issues/2113

Please let us know if that answears your question. If not, please could you add more details about the steps that you are performing ?

**Comment by svyatonik** [comment]
I've already responded in #2049. There are multiple accounts involved in this tiny command. For simplicity, let's only talk about following accounts:
- `source-signer` is simply an account who is signing `send_message` transaction at Millau. It pays transaction fee + it pays to relayers for message delivery and dispatch. This account lives ar Millau any only has Millau balance. So it pays in MLAU tokens;
- `target-signer` is the account at Rialto chain. It has RLT tokens. This is the account that will be used as the message dispatch origin;
- `5qTnmq6AGPoZVr29wDsgXhoLhpp3sCukSFXh7zGSaQtgCa1U` is the account at Rialto chain.

You have specified `transfer` subcommand, which means that the bridge message will be the encoded `pallet_balances::Call::transfer(<dispatch-origin>, <recipient>, <amount>)` call. This call will be dispatched at Rialto chain. Let's map it to our arguments:
- amount is `10000000000000` RLT tokens;
- recipient is `5qTnmq6AGPoZVr29wDsgXhoLhpp3sCukSFXh7zGSaQtgCa1U` Rialto account;
- dispatch origin is `target-signer`.

So the side effect of the dispatch will be (at Rialto):
- `5qTnmq6AGPoZVr29wDsgXhoLhpp3sCukSFXh7zGSaQtgCa1U` balance is increased by `10000000000000` RLT tokens;
- `target-signer` balance is decreased by `10000000000000` tokens.

---

In other words. with encoded-calls messaging you can't convert tokens and magically withdraw `10000000000000` MLAU tokens at one chain and add `10000000000000` RLT tokens to some account at other chain. The `transfer` command transfers balances between two accounts at the same (target) chain. The bridge allows to control that from the other (source) chain. That's it.

If you need to have some wrapped tokens, you may do that with encoded-calls messaging too. But you'll need to deal with assets pallet (not the basic balances pallets) and manually register assets wrapped there. Then you should be able to: (1) withdraw MLAU tokens and put it into some storage (or simply burn) (2) encode assets transfer call and send it to the other chain. This isn't trivial.
